org.eclipse.draw2d/src/org/eclipse/draw2d/graph/CompoundRankSorter.java

```diff
@@ -47,6 +47,13 @@ static class RowKey {
 
 		@Override
 		public boolean equals(Object obj) {
+			if (this == obj) {
```

Comment:
Can this class also be converted to a record?

Reply:
yes makes the code a bit cleaner